platform/machine/packet/cluster.go

```diff
@@ -90,37 +98,37 @@ func (pc *cluster) NewMachine(userdata *conf.UserData) (platform.Machine, error)
 		mach.publicIP = pc.flight.api.GetDeviceAddress(device, 4, true)
 		mach.privateIP = pc.flight.api.GetDeviceAddress(device, 4, false)
 		if mach.publicIP == "" || mach.privateIP == "" {
-			mach.Destroy()
+			pc.flight.api.DeleteDevice(mach.ID())
 			err = fmt.Errorf("couldn't find IP addresses for device")
 			continue // provisioning error
 		}
 
 		dir := filepath.Join(pc.RuntimeConf().OutputDir, mach.ID())
```

Comment:
```suggestion
		dir := filepath.Join(pc.RuntimeConf().OutputDir, mach.ID())
```
This here will overwrite the existing file, we should not use the machine ID here and rather need a suffix for it. I would like to see this directly be part of the `.ID()` function to be able to distinguish the machines reuse when tests log their ID. The `vmname` above already is a random string that we can append in the machine's `ID()` function.

Below for the deletion the real machine ID needs to be used and I suggest that this becomes a function on the machine struct.

Reply:
I think it's alright - indeed instances will have the same ID but they won't be in the same folder:
```
$ tree _kola_temp/packet-latest
_kola_temp/packet-latest
├── cl.basic
│   └── a279e8b2-14f4-451c-acd5-8705df9e70b5
│       ├── console.txt
│       ├── journal-raw.txt.gz
│       ├── journal.txt
│       └── user-data
├── cl.internet
│   └── a279e8b2-14f4-451c-acd5-8705df9e70b5
│       ├── console.txt
│       ├── journal-raw.txt.gz
│       ├── journal.txt
│       └── user-data
```

There is no chance from the same flight to have instances recycled under the same cluster I think.

Or I miss a usecase ?

Reply:
Good, I don't see a test that destroys the machine/cluster it created and would create a new one afterwards. I think we are safe as long as this is the case.